src/keylessBackup/SignInWithEmail.tsx

```diff
@@ -1,31 +1,60 @@
 import React from 'react'
 import { useTranslation } from 'react-i18next'
 import { SafeAreaView, ScrollView, StyleSheet, Text } from 'react-native'
+import { useAuth0 } from 'react-native-auth0'
 import { useDispatch } from 'react-redux'
 import { KeylessBackupEvents } from 'src/analytics/Events'
 import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
 import Button, { BtnSizes, BtnTypes } from 'src/components/Button'
 import GoogleIcon from 'src/icons/Google'
 import Times from 'src/icons/Times'
-import { googleSignInStarted } from 'src/keylessBackup/slice'
+import { googleSignInCompleted } from 'src/keylessBackup/slice'
+import { KeylessBackupFlow } from 'src/keylessBackup/types'
 import { emptyHeader } from 'src/navigator/Headers'
-import { navigateHome } from 'src/navigator/NavigationService'
+import { navigate } from 'src/navigator/NavigationService'
 import { Screens } from 'src/navigator/Screens'
 import { TopBarIconButton } from 'src/navigator/TopBarButton'
 import fontStyles from 'src/styles/fonts'
+import Logger from 'src/utils/Logger'
+
+const TAG = 'keylessBackup/SignInWithEmail'
 
 function onPressCancel() {
-  ValoraAnalytics.track(KeylessBackupEvents.sign_in_with_email_screen_cancel)
-  navigateHome({ params: { initialScreen: Screens.Settings } })
+  ValoraAnalytics.track(KeylessBackupEvents.cab_sign_in_with_email_screen_cancel)
+  navigate(Screens.SetUpKeylessBackup)
 }
 
+// TODO: support recovery flow
 function SignInWithEmail() {
   const { t } = useTranslation()
   const dispatch = useDispatch()
+  const { authorize, getCredentials, clearCredentials } = useAuth0()
+
+  const onPressGoogle = async () => {
```

Comment:
I noticed it took a solid 1s to navigate to the next screen after the Google UI went away. Should we show a spinner in the "Sign in with Google" button or at least disable it while this function is running?

Reply:
yeah good call, was talking to Nitya earlier today about it. Will make a follow up PR to include a spinner?